warning: in the working copy of 'UAM-Toy-Env/eval.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'UAM-Toy-Env/tests/uam_toy_environment-v0.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'UAM-Toy-Env/train.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'UAM-Toy-Env/wandb_logger/base_logger.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'UAM-Toy-Env/wandb_logger/logger_util.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'UAM-Toy-Env/wandb_logger/wandb_logger.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_001520-7shyz8ow/files/config.yaml', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_001520-7shyz8ow/files/requirements.txt', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_001520-7shyz8ow/files/wandb-metadata.json', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_002559-hx5khlpt/files/code/UAM-Toy-Env/train.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_002559-hx5khlpt/files/config.yaml', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_002559-hx5khlpt/files/requirements.txt', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_002559-hx5khlpt/files/wandb-metadata.json', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_003312-zus9ahlb/files/code/UAM-Toy-Env/train.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_003312-zus9ahlb/files/config.yaml', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_003312-zus9ahlb/files/requirements.txt', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_003312-zus9ahlb/files/wandb-metadata.json', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_005753-w8hbs4w8/files/code/UAM-Toy-Env/train.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_005753-w8hbs4w8/files/config.yaml', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_005753-w8hbs4w8/files/requirements.txt', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_005753-w8hbs4w8/files/wandb-metadata.json', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_010051-1mhhwq9x/files/code/UAM-Toy-Env/train.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_010051-1mhhwq9x/files/config.yaml', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_010051-1mhhwq9x/files/requirements.txt', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_010051-1mhhwq9x/files/wandb-metadata.json', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011031-msdsacuq/files/code/UAM-Toy-Env/train.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011031-msdsacuq/files/config.yaml', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011031-msdsacuq/files/requirements.txt', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011031-msdsacuq/files/wandb-metadata.json', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011218-yl4adgwk/files/code/UAM-Toy-Env/train.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011218-yl4adgwk/files/config.yaml', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011218-yl4adgwk/files/requirements.txt', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011218-yl4adgwk/files/wandb-metadata.json', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011425-743vjd17/files/code/UAM-Toy-Env/train.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011425-743vjd17/files/config.yaml', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011425-743vjd17/files/requirements.txt', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011425-743vjd17/files/wandb-metadata.json', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011709-85m6ilfs/files/code/UAM-Toy-Env/train.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011709-85m6ilfs/files/config.yaml', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011709-85m6ilfs/files/requirements.txt', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011709-85m6ilfs/files/wandb-metadata.json', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011839-uxhwhrr0/files/code/UAM-Toy-Env/train.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011839-uxhwhrr0/files/config.yaml', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011839-uxhwhrr0/files/requirements.txt', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_011839-uxhwhrr0/files/wandb-metadata.json', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_012229-fsv4zapd/files/code/UAM-Toy-Env/train.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_012229-fsv4zapd/files/config.yaml', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_012229-fsv4zapd/files/requirements.txt', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_012229-fsv4zapd/files/wandb-metadata.json', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_012422-ecukq2ji/files/code/UAM-Toy-Env/train.py', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_012422-ecukq2ji/files/config.yaml', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_012422-ecukq2ji/files/requirements.txt', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'wandb/run-20250611_012422-ecukq2ji/files/wandb-metadata.json', LF will be replaced by CRLF the next time Git touches it
diff --git a/UAM-Toy-Env/train.py b/UAM-Toy-Env/train.py
index 9e866bb3..d44cc22b 100644
--- a/UAM-Toy-Env/train.py
+++ b/UAM-Toy-Env/train.py
@@ -15,7 +15,7 @@ from wandb.integration.sb3 import WandbCallback
 sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 from uam_toy_environment.environ.uam_toy_environment import UAMToyEnvironment
 
-total_timesteps = 1000000  # Define the total timesteps for training.
+total_timesteps = 2e5  # Define the total timesteps for training.
 # Initialize WandB for logging.
 wandb.init(
     project="uam-toy-project",  # Change this to your project name
@@ -65,7 +65,7 @@ model_PPO1 = PPO(
 # )
 # Train the model for a specified number of timesteps.
 model_PPO1.learn(
-    total_timesteps=total_timesteps,
+    total_timesteps=int(total_timesteps),
     tb_log_name="test run",
     progress_bar=True,
     callback=WandbCallback()
diff --git a/UAM-Toy-Env/uam_toy_environment/environ/uam_toy_environment.py b/UAM-Toy-Env/uam_toy_environment/environ/uam_toy_environment.py
index bcb64924..a056cadb 100644
--- a/UAM-Toy-Env/uam_toy_environment/environ/uam_toy_environment.py
+++ b/UAM-Toy-Env/uam_toy_environment/environ/uam_toy_environment.py
@@ -238,8 +238,8 @@ class UAMToyEnvironment(gym.Env):
             self.num_vertiports - (drone_starting_vertiport + 1)
         ]
         # if moving closer to vertiport, positive reward, negative if moving farther away
-        reward_goal = (10. * (-np.linalg.norm(next_dist) + np.linalg.norm(initial_dist))) / self.grid_size
-        # reward_goal = -np.linalg.norm(next_dist)
+        # reward_goal = (10. * (-np.linalg.norm(next_dist) + np.linalg.norm(initial_dist))) / self.grid_size
+        reward_goal = -np.linalg.norm(next_dist)
 
         # kp, ki, kd = 1, 1, 1 # Proportional, integral, and derivative gains for the reward function
         # reward_goal = -kp * np.linalg.norm(initial_dist) - ki * np.linalg.norm(next_dist - initial_dist) * self.dt - kd * np.linalg.norm(next_dist - initial_dist) / self.dt
@@ -247,6 +247,7 @@ class UAMToyEnvironment(gym.Env):
         agent_collision = 0.0
         obstacle_collision = 0.0
         out_of_bounds = 0.0
+        approaching_vertiport = 0.0
         vertiport_reached = 0.0
         time = 0.0
         for i in range(self.num_drones):
@@ -265,7 +266,7 @@ class UAMToyEnvironment(gym.Env):
                 self.obstacles_pos[i],
                 self.S_o,
             ):
-                obstacle_collision += -5.0
+                obstacle_collision += -100.0
         for drone in self.drones:
             any_out_of_bounds = (
                 self.drone_pos[drone][0] < 0
@@ -273,7 +274,7 @@ class UAMToyEnvironment(gym.Env):
                 or self.drone_pos[drone][1] < 0
                 or self.drone_pos[drone][1] > self.grid_size)
             if any_out_of_bounds:
-                out_of_bounds += -5.0
+                out_of_bounds += -100.0
         for drone in self.drones:
             if self._is_overlapping(
                 self.drone_pos[drone],
@@ -282,7 +283,7 @@ class UAMToyEnvironment(gym.Env):
                 self.S_v + 2,
             ):
             # if all(self.drone_pos[drone] == self.vertiports_loc[1 if self.drone_vertiport[drone] == 0 else 0]):
-                vertiport_reached += 50.0
+                approaching_vertiport += 100.0
                 # time += (self.max_steps - self.time_step) * 1.0e4
             if self._is_overlapping(
                 self.drone_pos[drone],
@@ -290,15 +291,14 @@ class UAMToyEnvironment(gym.Env):
                 self.vertiports_loc[1 if self.drone_vertiport[drone] == 0 else 0],
                 0.5 * self.S_v,
             ):
-                vertiport_reached += 50.0 * 0
+                vertiport_reached += 500.0
                 # time += (self.max_steps - self.time_step) * 1.0e4
                 print("Reached vertiport!")
             
         # if self.time_step >= self.max_steps:
         #     time += -1000.0
 
-
-        reward = 1 * reward_goal + agent_collision + obstacle_collision + out_of_bounds + vertiport_reached + time
+        reward = 1 * reward_goal + agent_collision + obstacle_collision + out_of_bounds + approaching_vertiport + vertiport_reached + time
 
         return reward
 
diff --git a/uam_toy.zip b/uam_toy.zip
index 156c1bb0..0215b16b 100644
Binary files a/uam_toy.zip and b/uam_toy.zip differ
diff --git a/wandb/latest-run b/wandb/latest-run
index 83be4d53..41b7b9de 120000
--- a/wandb/latest-run
+++ b/wandb/latest-run
@@ -1 +1 @@
-run-20250611_012422-ecukq2ji
\ No newline at end of file
+run-20250611_125215-ax7qom3y
\ No newline at end of file
